mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -231,4 +242,57 @@ public void searchShouldNotReturnResultsFromOtherUsers() throws Exception {
         assertThat(metaDatas.get(0).getPath()).isEqualTo(MailboxPath.inbox(session));
     }
 
+    @ContractTest
+    public void updateAnnotationsShouldUpdateStoredAnnotation() throws MailboxException {
+        MailboxPath inbox = setUpMailbox();
+        mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(PRIVATE_ANNOTATION));
+
+        mailboxManager.updateAnnotations(inbox, session, Lists.newArrayList(PRIVATE_ANNOTATION_UPDATE));
+        assertThat(mailboxManager.getAllAnnotations(inbox, session)).isEqualTo(ImmutableList.of(PRIVATE_ANNOTATION_UPDATE));
+    }
+
+    @ContractTest
+    public void updateAnnotationsShouldDeleteAnnotationWithNilValue() throws BadCredentialsException, MailboxException {
+        MailboxPath inbox = setUpMailbox();
+        mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(PRIVATE_ANNOTATION));
+
+        mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(MailboxAnnotation.nil("/private/comment")));
+        assertThat(mailboxManager.getAllAnnotations(inbox, session)).isEmpty();
+    }
+
+    @ContractTest
+    public void getAnnotationsShouldReturnEmptyForNonStoredAnnotation() throws BadCredentialsException, MailboxException {
+        MailboxPath inbox = setUpMailbox();
+
+        assertThat(mailboxManager.getAllAnnotations(inbox, session)).isEmpty();
+    }
+
+    @ContractTest
+    public void getAnnotationsShouldRetrieveStoredAnnotations() throws BadCredentialsException, MailboxException {
+        MailboxPath inbox = setUpMailbox();
+
+        mailboxManager.updateAnnotations(inbox, session, ANNOTATIONS);
+
+        assertThat(mailboxManager.getAllAnnotations(inbox, session)).isEqualTo(ANNOTATIONS);
+    }
+
+    @ContractTest
+    public void getAnnotationsShouldRetrieveStoresAnnotationsByKeys() throws BadCredentialsException, MailboxException {
+        MailboxPath inbox = setUpMailbox();
+
+        mailboxManager.updateAnnotations(inbox, session, ANNOTATIONS);
+
+        assertThat(mailboxManager.getAnnotationsByKeys(inbox, session, ImmutableList.of("/private/comment")))
+            .isEqualTo(ImmutableList.of(PRIVATE_ANNOTATION));
+    }
+
+    private MailboxPath setUpMailbox() throws MailboxException {
+        Assume.assumeTrue(mailboxManager.getSupportedMailboxCapabilities().contains(MailboxCapabilities.Annotation));
```

Comment:
Yes, there are more than 2 places used the same checking like that. But should I just refactor my code or should I refactor all impact code?